tests/sdk/data/test_dpsu.py

```diff
@@ -0,0 +1,34 @@
+import os
+import subprocess
+
+import pandas as pd
+import math
+import pytest
+
+from opendp.whitenoise.data.dpsu import preprocess_df_from_query, policy_laplace, dpsu_df
+from opendp.whitenoise.metadata import CollectionMetadata
+from opendp.whitenoise.sql import PrivateReader, PandasReader
+from opendp.whitenoise.sql.parse import QueryParser
+
+git_root_dir = subprocess.check_output("git rev-parse --show-toplevel".split(" ")).decode("utf-8").strip()
+
+meta_path = os.path.join(git_root_dir, os.path.join("service", "datasets", "PUMS.yaml"))
+csv_path = os.path.join(git_root_dir, os.path.join("service", "datasets", "PUMS.csv"))
+
+class TestDPSU:
+    def test_preprocess_df_from_query(self):
+        query = "SELECT married, educ FROM PUMS.PUMS GROUP BY married, educ"
+        df = preprocess_df_from_query(meta_path, csv_path, query, "PUMS.PUMS")
+
+        assert df is not None
+        assert(len(df["group_cols"][0]) == 2)
+
+    def test_dpsu_df(self):
+        original_df = pd.read_csv(csv_path)
+        query = "SELECT married, educ FROM PUMS.PUMS GROUP BY married, educ"
+        df = preprocess_df_from_query(meta_path, csv_path, query, "PUMS.PUMS")
```

Comment:
What is our intended analyst experience?
Would it be something like:
dpsu_df = run_dpsu(df, metadata, query, eps, max_contrib, delta=some constant/based off size of df)

Can max contrib be defaulted @joshua-oss ?

Reply:
per meeting, default max_contrib to 5, default delta and see if you can get table name from metadata